src/Compilers/CSharp/Test/Symbol/Symbols/ExtendedPartialMethodsTests.cs

```diff
@@ -0,0 +1,552 @@
+
+            comp = CreateCompilation(text1, parseOptions: TestOptions.RegularWithExtendedPartialMethods);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void New_LangVersion()
+        {
+            const string text1 = @"
+class C
+{
+    internal void M1() { }
+}
+
+partial class D : C
+{
+    new partial void M1();
+    new partial void M1() { }
+}";
+            var comp = CreateCompilation(text1);
+            comp.VerifyDiagnostics(
+                // (9,22): error CS8652: The feature 'extended partial methods' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //     new partial void M1();
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "M1").WithArguments("extended partial methods").WithLocation(9, 22),
+                // (10,22): error CS8652: The feature 'extended partial methods' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //     new partial void M1() { }
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "M1").WithArguments("extended partial methods").WithLocation(10, 22));
+
+            comp = CreateCompilation(text1, parseOptions: TestOptions.RegularWithExtendedPartialMethods);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void New_NoImpl()
+        {
+            const string text1 = @"
+class C
+{
+    internal void M1() { }
+}
+
+partial class D : C
+{
+    new partial void M1();
+}";
+            var comp = CreateCompilation(text1, parseOptions: TestOptions.RegularWithExtendedPartialMethods);
+            comp.VerifyDiagnostics(
+                // (9,22): error CS9053: Partial method D.M1() must have an implementation part because it is has a 'virtual', 'override', 'sealed', or 'new', or 'extern' modifier.
+                //     new partial void M1();
+                Diagnostic(ErrorCode.ERR_PartialMethodWithExtendedModMustHaveImplementation, "M1").WithArguments("D.M1()").WithLocation(9, 22)
+            );
+        }
+
+        [Fact]
+        public void New_Mismatch_01()
+        {
+            const string text1 = @"
+class C
+{
+    internal void M1() { }
+}
+
+partial class D : C
+{
+    new partial void M1();
+    partial void M1() { }
+}";
+            var comp = CreateCompilation(text1, parseOptions: TestOptions.RegularWithExtendedPartialMethods);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void New_Mismatch_02()
+        {
+            const string text1 = @"
+class C
+{
+    internal void M1() { }
+}
+
+partial class D : C
+{
+    partial void M1();
+    new partial void M1() { }
+}";
+            var comp = CreateCompilation(text1, parseOptions: TestOptions.RegularWithExtendedPartialMethods);
+            comp.VerifyDiagnostics();
+        }
+    }
```

Comment:
>} [](start = 4, length = 1)

Please include a test with the same modifiers in definition and implementation parts but different order.

Reply:
Added test `SealedOverride_Reordered`.